### ISSUE TITLE
Start-Process: Create a new window without stealing focus

### ISSUE THREAD

**Issue by UselessGuru** [PowerShell/PowerShell#24387]
### Summary of the new feature / enhancement

Summary of the new feature / enhancement
I would like to start a new process with start-process, but so that the new process would NOT steal focus.

start-process -WindowStyle does not have an option to do so.

https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/Invoke-CreateProcess.ps1
has exactly what I am looking for.

### Proposed technical implementation details (optional)

Proposed technical implementation details (optional)
Add a new parameter, e.g. '-ShowWindow [nCmdShow value]' to Start-Process that takes any of the nCmdShow values listed in

https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showwindow

**Comment by rhubarb-geek-nz** [comment]
It is interesting that your code example is 9 years old, and I am sure it still works!

Now PSGallery or similar is the best place for this kind of code.

PowerShell 7 is based on dotnet (the SDK formally known as dotnet core) and for whatever reasons [ProcessWindowsStyle](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processwindowstyle) is an enum with only four options and this enum is then used to populate `lpStartupInfo.wShowWindow`

Now you could add more code to add more options but it is forever a losing race...

- Windows will add more options
- It won't get back ported to previous versions of PowerShell
- You have increased the divergence from the relative simplicity of the cross platform dotnet API.
- It is more platform specific code that means nothing on Linux, FreeBSD or macOS.

So if this is in an external PowerShell module then

- You can run this on _any_ version of PowerShell
- There is no backward compatibility problem
- As Windows adds more options, just this module has to change.

As a cross platform "shell" ( deliberately in inverted commas ), PowerShell Core should really have had nothing to do with windowing systems, alas it is still dragging around memories of Windows 95 with Shell Execute verbs.

So yes, an excellent idea, but put in in a PSGallery module then everyone can benefit now.

**Comment by UselessGuru** [comment]
>> It is interesting that your code example is 9 years old, and I am sure it still works!
Yes it still works well - all the way to the latest Win 11 24h2 :-)

>> Now PSGallery or similar is the best place for this kind of code.
I do not want to push the code snipped to anywhere.
Much more I am looking for an easy way to spawn a new process with start-process without it stealing focus.

>>and for whatever reasons [ProcessWindowsStyle](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processwindowstyle) is an enum with only four options and this enum is then used to populate lpStartupInfo.wShowWindow
It is interesting to see that https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processwindowstyle?view=net-8.0 only shows 4 values. 

But https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/Invoke-CreateProcess.ps1 has much more combiniations for wShowWindow:

     -ShowWindow        Window display flags:
                          0x0000 (SW_HIDE)
                          0x0001 (SW_SHOWNORMAL)
                          0x0001 (SW_NORMAL)
                          0x0002 (SW_SHOWMINIMIZED)
                          0x0003 (SW_SHOWMAXIMIZED)
                          0x0003 (SW_MAXIMIZE)
                          0x0004 (SW_SHOWNOACTIVATE)
                          0x0005 (SW_SHOW)
                          0x0006 (SW_MINIMIZE)
                          0x0007 (SW_SHOWMINNOACTIVE)
                          0x0008 (SW_SHOWNA)
                          0x0009 (SW_RESTORE)
                          0x000A (SW_SHOWDEFAULT)
                          0x000B (SW_FORCEMINIMIZE)
                          0x000B (SW_MAX)

$StartupInfo.wShowWindow = $ShowWindow # StartupInfo.ShowWindow

And they still work.
So I was hoping that there is an easy way to pass any of these to start-process.

**Comment by rhubarb-geek-nz** [comment]
> So I was hoping that there is an easy way to pass any of these to start-process.

There is an easy way, define another enum and another parameter and then 

1. have two parameters which control the same value (wShowWindow)

2. define which takes precedence when both are provided

3. when your script runs on any prior version of PowerShell it errors with

```
Start-Process: A parameter cannot be found that matches parameter name 'ShowWindow'.
```

So it more about the difference between "can" and "should", and the overall value it provides.

**Comment by UselessGuru** [comment]
@rhubarb-geek-nz 

You speak in riddles ;-)

> have two parameters which control the same value (wShowWindow)
> define which takes precedence when both are provided

How would I do this?

**Comment by rhubarb-geek-nz** [comment]
> How would I do this?

I presumed you were expecting Start-Process to be changed. It already accepts 

```
-WindowStyle <ProcessWindowStyle>
```
and you want it to also accept

```
-ShowWindow <ShowWindowStyle>
```

I count that has two parameters, both of which were intended to set the value of wShowWindow in the STARTUPINFO.

I was describing the coding change and the effect of that change,

**Comment by rhubarb-geek-nz** [comment]
> However I wouldn't know how to achieve this myself.

Fortunately you have already take the correct first steps. This issue is marked for Triage and Issue-Enhancement so it is in the queue of things to get looked by the relevant working group. These initial discussions are also useful because they flesh out a little more detail about both expectations and possible implementation.

**Comment by TobiasPSP** [comment]
The Cmdlet Working Group discussed this. Start-Process is designed to work cross-platform. It uses the managed .NET API which currently does not expose the requested feature. At the current time, community modules like the one referenced (https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/Invoke-CreateProcess.ps1) use the Windows-specific API and are a good workaround for Windows users. Once .NET natively adds this feature, we'll look into integrating it.
